apps/mark/frontend/src/pages/idle_page.tsx

```diff
@@ -1,14 +1,19 @@
 import { IdlePage as MarkFlowIdlePage } from '@votingworks/mark-flow-ui';
-import { useContext } from 'react';
+import { useCallback, useContext } from 'react';
 import { BallotContext } from '../contexts/ballot_context';
+import { endCardlessVoterSession } from '../api';
 
 export function IdlePage(): JSX.Element {
-  const { endVoterSession, resetBallot } = useContext(BallotContext);
+  const endCardlessVoterSessionMutation = endCardlessVoterSession.useMutation();
+  // mutate function, unlike mutation object, does not change identity each render
+  const endCardlessVoterSessionMutate = endCardlessVoterSessionMutation.mutate;
+  const { resetBallot } = useContext(BallotContext);
 
-  async function reset() {
-    await endVoterSession();
-    resetBallot();
-  }
+  const reset = useCallback(() => {
+    endCardlessVoterSessionMutate(undefined, {
+      onSuccess: () => resetBallot(),
+    });
+  }, [endCardlessVoterSessionMutate, resetBallot]);
```

Comment:
The `mutate` method has a consistent identity so doesn't cause runaway `useEffect`s like were apparently botching that test. 

I think we're still using mutations as dependencies in a lot of places. Annoyingly, the lint rule doesn't let you specify properties in the dependency array, some discussion here: https://github.com/facebook/react/issues/16265. That lint rule is otherwise very helpful, so we don't want to disable it often. Which is a shame, because I think passing `someMutation.mutate` to the dependency array is more idiomatic than what I do above.

Thoughts?

Reply:
How would you access the `.mutate` method inside the `useCallback` without also depending on the mutation object as well? Wouldn't the callback function close over the mutation object?

Reply:
The callback function would close over the mutation object but, since the identify of `.mutate` is consistent, even if the mutation object we're accessing is old the `.mutate` reference should still be correct? 

But we can't allow such a think within the react-hooks lint rule, unless there's a way to make programmatic carve-outs to rules (like post-processing to ignore certain cases). I assume that such a thing is possible within `eslint`, but maybe complex.

Reply:
Ah yes, I see how that would work. I don't mind the bit of extra boilerplate to pull it out into a separate variable. A little annoying, but that's ok.